WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/AddProductName/AddProductNameWithAIView.swift

```diff
@@ -0,0 +1,256 @@
+                        .foregroundColor(Color(.secondaryLabel))
+                        .bodyStyle()
+                }
+                .padding(.bottom, Layout.titleBlockBottomPadding)
+
+                VStack(alignment: .leading, spacing: Layout.titleBlockSpacing) {
+                    VStack(alignment: .leading, spacing: Layout.editorBlockSpacing) {
+                        Text(Localization.productName)
+                            .foregroundColor(Color(.label))
+                            .subheadlineStyle()
+
+                        textEditor
+                    }
+
+                    usePackagePhoto
+                }
+            }
+            .padding(insets: Layout.insets)
+        }
+        .safeAreaInset(edge: .bottom) {
+            VStack {
+                // CTA to continue to next screen.
+                continueButton
+                    .padding()
+            }
+            .background(Color(uiColor: .systemBackground))
+        }
+    }
+}
+
+private extension AddProductNameWithAIView {
+    @ViewBuilder
+    var textEditor: some View {
+        if #available(iOS 16.0, *) {
+            VStack(spacing: 0) {
+                TextField(Localization.placeholder, text: $viewModel.productNameContent, axis: .vertical)
+                .lineLimit(3...)
+                .bodyStyle()
+                .foregroundColor(.secondary)
+                .textFieldStyle(.plain)
+                .focused($editorIsFocused)
+                .padding(insets: Layout.messageContentInsets)
+
+                Divider()
+                    .frame(height: Layout.dividerHeight)
+                    .foregroundColor(Color(.separator))
+
+                suggestNameField
+                    .padding(insets: Layout.suggestButtonInsets)
+            }
+            .overlay(
+                RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(editorIsFocused ? Color(.brand) : Color(.separator))
+            )
+        } else {
+            ZStack(alignment: .topLeading) {
+                VStack(spacing: 0) {
+                    TextEditor(text: $viewModel.productNameContent)
+                        .bodyStyle()
+                        .foregroundColor(.secondary)
+                        .padding(insets: Layout.messageContentInsets)
+                        .frame(minHeight: Layout.minimumEditorHeight, maxHeight: .infinity)
+                        .focused($editorIsFocused)
+
+                    Divider()
+                        .frame(height: Layout.dividerHeight)
+                        .foregroundColor(Color(.separator))
+
+                    suggestNameField
+                        .padding(insets: Layout.suggestButtonInsets)
+                }
+                .overlay(
+                    RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(editorIsFocused ? Color(.brand) : Color(.separator))
+                )
+
+                // Placeholder text
+                placeholderText
+            }
```

Comment:
💡 I found a solution to fix the frame size of this view when the placeholder text is large. Since `ZStack` always fit the largest view it contains, we should use one to contain both the text view and the placeholder text, and it should be contained in the `VStack` together with the divider and the button:

```suggestion
            VStack(spacing: 0) {
                ZStack(alignment: .topLeading) {
                    TextEditor(text: $viewModel.productNameContent)
                        .bodyStyle()
                        .foregroundColor(.secondary)
                        .padding(insets: Layout.messageContentInsets)
                        .frame(minHeight: Layout.minimumEditorHeight, maxHeight: .infinity)
                        .focused($editorIsFocused)
                        
                        // Placeholder text
                       placeholderText
                   }

                   Divider()
                       .frame(height: Layout.dividerHeight)
                       .foregroundColor(Color(.separator))

                  suggestNameField
                      .padding(insets: Layout.suggestButtonInsets)
             }
             .overlay(
                    RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(editorIsFocused ? Color(.brand) : Color(.separator))
              )
```

This way, we will not need to switch to `TextField` for iOS 16 and above.

Reply:
Nice fix. 🎖️ Added this in https://github.com/woocommerce/woocommerce-ios/pull/10723/commits/8862b943afa4887b077fb2cb3bdfaac38a1950e5